x-pack/test/functional/apps/graph/graph.ts

```diff
@@ -0,0 +1,176 @@
+      // wait for the saved object to be loaded
+      // TODO this race condition will be removed with eui-ification
+      // of graph bar
+      await PageObjects.common.sleep(1000);
+      // select fields url.parts, url, params and src
+      await PageObjects.graph.addField('url.parts');
+      await PageObjects.graph.addField('url');
+      await PageObjects.graph.addField('params');
+      await PageObjects.graph.addField('src');
+      await PageObjects.graph.query('admin');
+      await PageObjects.common.sleep(8000);
+    }
+
+    it('should show correct node labels', async function() {
+      await buildGraph();
+      const { nodes } = await PageObjects.graph.getGraphObjects();
+      const circlesText = nodes.map(({ label }) => label);
+      expect(circlesText.length).to.equal(expectedNodes.length);
+      expect(circlesText).to.eql(expectedNodes);
+    });
+
+    it('should show correct connections', async function() {
+      const epsilon = Number.EPSILON;
+      const expectedConnectionCount = Object.values(expectedConnectionWidth)
+        .map(connections => Object.values(connections).length)
+        .reduce((acc, n) => acc + n, 0);
+      const { edges } = await PageObjects.graph.getGraphObjects();
+      expect(edges.length).to.be(expectedConnectionCount);
+      edges.forEach(edge => {
+        const from = edge.sourceNode.label!;
+        const to = edge.targetNode.label!;
+        // fuzzy matching to take floating point rounding issues into account
+        expect(expectedConnectionWidth[from][to]).to.be.within(
+          edge.width - epsilon,
+          edge.width + epsilon
+        );
+      });
+    });
+
+    it('should save Graph workspace', async function() {
+      const graphExists = await PageObjects.graph.saveGraph(graphName);
+      expect(graphExists).to.eql(true);
+    });
+
+    // open the same graph workspace again and make sure the results are the same
+    it('should open Graph workspace', async function() {
+      await PageObjects.graph.openGraph(graphName);
+      const { nodes } = await PageObjects.graph.getGraphObjects();
+      const circlesText = nodes.map(({ label }) => label);
+      expect(circlesText.length).to.equal(expectedNodes.length);
+      expect(circlesText).to.eql(expectedNodes);
+    });
+
+    it('should create new Graph workspace', async function() {
+      await PageObjects.graph.newGraph();
+      const { nodes, edges } = await PageObjects.graph.getGraphObjects();
+      expect(nodes).to.be.empty();
+      expect(edges).to.be.empty();
+    });
+
+    it('should show venn when clicking a line', async function() {
+      await buildGraph();
+      const { edges } = await PageObjects.graph.getGraphObjects();
+
+      const blogAdminBlogEdge = edges.find(
+        ({ sourceNode, targetNode }) =>
+          sourceNode.label === '/blog/wp-admin/' && targetNode.label === 'blog'
+      )!;
+
+      await PageObjects.graph.isolateEdge(blogAdminBlogEdge);
```

Comment:
I'm glad you're working on this Joe!  One thought here.  I was really not happy with the way I wrote the original graph test in this area.  I put the whole creation of the graph in a loop so I could keep trying to click a unique edge and it was often blocked by a node's circle.  But I think would have been better to send the click event to that element directly.  Then (I think) it wouldn't have mattered if there was something on top of it.  Something like;
```javascript
const elem = await testSubjects.find('timepickerGoButton');
await remote.execute('arguments[0].click();', [elem]);
```

Reply:
I considered this, but I decided against it for two reasons:
* It's not testing anymore against the case that the edge is actually unclickable (e.g. because another element is shadowing it or the visualization itself is outside of the viewport), which should be caught
* This approach also covers some other functions of the interface as well (selecting and deleting nodes).